MyCalculator/ViewController.swift

```diff
@@ -58,26 +88,61 @@ class ViewController: UIViewController {
                 brain.performOperations(mathmaticalSymbol)
             }
             
-            if let result = brain.result{
-                displayValue = result
-                displayDescription.text = brain.getDescription
+            if brain.result != nil{
+                displayValueToUpdate = brain.evaluate(using: variablesDictionary)
             }
         }
-
     }
     
     @IBAction func clear(_ sender: UIButton) {
         brain.clear()
         userIsInMiddleOfTyping = false
         displayValue = 0
         displayDescription.text = " "
+        variablesDictionary = nil
+        displayM.text = " M = 0 "
     }
     
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view, typically from a nib.
     }
 
+    // →M
+    
+    @IBAction func getM(_ sender: UIButton) {
+        // M: current value of the display
+        variablesDictionary = ["M": displayValue]
```

Comment:
i know, this is the requirement in the task.
first, i just override the value of M,
but than i notice that the task says to drop the whole dictionary and use a new one.

Reply:
right.. so in `clear()` I would do something like:
`variablesDictionary = [String: Double]()`

instead of

`variablesDictionary = nil`

And in `getM` just do 
`variablesDictionary['M'] = displayValue`

Reply:
agree, i'll change it